pkgs/development/libraries/science/math/suitesparse/default.nix

```diff
@@ -4,32 +4,35 @@
 , blas, lapack
 , metis
 , fixDarwinDylibNames
-, gnum4
+, gmp
+, mpfr
 , enableCuda ? false
 , cudatoolkit
 }:
 
 stdenv.mkDerivation rec {
   pname = "suitesparse";
-  version = "5.7.2";
+  version = "5.8.1";
 
   outputs = [ "out" "dev" "doc" ];
 
   src = fetchFromGitHub {
     owner = "DrTimothyAldenDavis";
     repo = "SuiteSparse";
     rev = "v${version}";
-    sha256 = "1imndff7yygjrbbrcscsmirdi8w0lkwj5dbhydxmf7lklwn4j3q6";
+    sha256 = "0qjlyfxs8s48rs63c2fzspisgq1kk4bwkgnhmh125hgkdhrq2w1c";
   };
 
   nativeBuildInputs = [
-    gnum4
   ] ++ stdenv.lib.optional stdenv.isDarwin fixDarwinDylibNames;
 
-  buildInputs = [
+  # Use compatible indexing for lapack and blas used
+  buildInputs = assert (blas.isILP64 == lapack.isILP64); [
```

Comment:
@jtojnar you mean like this?

Reply:
Yes, this should work.

See https://github.com/NixOS/nixpkgs/pull/94892#discussion_r471110250 for explanation. I thought there was an issue but cannot find it.